.github/workflows/build-main.yml

```diff
@@ -0,0 +1,66 @@
+name: Build
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+
+env:
+  BALLERINA_DISTRIBUTION_VERSION: 2201.8.0-rc1   # Update this with the latest Ballerina version
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    if: github.repository_owner == 'ballerina-platform'
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v3
+      - name: Set up JDK 17
+        uses: actions/setup-java@v3
+        with:
+          distribution: adopt
+          java-version: 17.0.7
+      - name: Ballerina Version
+        id: ballerina-version
+        run: |
+          BALLERINA_VERSION=$(cut -d '-' -f1 <<< "${{ env.BALLERINA_DISTRIBUTION_VERSION }}")
+          echo "::set-output name=version::$BALLERINA_VERSION"
+      - name: Install Ballerina
+        run: |
+          wget https://github.com/ballerina-platform/ballerina-distribution/releases/download/v${{ env.BALLERINA_DISTRIBUTION_VERSION }}/ballerina-${{ steps.ballerina-version.outputs.version }}-swan-lake-linux-x64.deb
```

Comment:
Lets use setup ballerina action or github action

Reply:
Addressed the suggestions. I used setup ballerina since we need to specify the ballerina version. I have changed the workflow to pickup ballerina version from `workflow_dispatch` inputs as well.